tests/ui/needless_take_by_value.stderr

```diff
@@ -0,0 +1,16 @@
+error: this function taking a value by value, but only using them by reference
+ --> $DIR/needless_take_by_value.rs:9:23
+  |
+9 | fn foo<T: Default>(v: Vec<T>, w: Vec<T>, mut x: Vec<T>, y: Vec<T>) -> Vec<T> {
+  |                       ^^^^^^
+  |
+note: lint level defined here
+ --> $DIR/needless_take_by_value.rs:4:9
+  |
+4 | #![deny(needless_take_by_value)]
+  |         ^^^^^^^^^^^^^^^^^^^^^^
+help: consider taking a reference instead
+  | fn foo<T: Default>(v: &Vec<T>, w: Vec<T>, mut x: Vec<T>, y: Vec<T>) -> Vec<T> {
```

Comment:
This looks like a problem quite frustrating for users and I would like this lint to be aware of `ptr_arg`.